Source/GRChomboCore/ChomboParameters.hpp

```diff
@@ -281,6 +291,77 @@ class ChomboParameters
         pout() << endl;
     }
 
+    void check_params()
+    {
+        check_parameter("L", L, L > 0.0, "must be > 0.0");
+        check_parameter("max_level", max_level, max_level >= 0, "must be >= 0");
+        // the following check assumes you will be taking some fourth (or
+        // higher) order one-sided derivatives
+        check_parameter("num_ghosts", num_ghosts, num_ghosts >= 3,
+                        "must be >= 3");
```

Comment:
This is in anticipation of me adding sixth order derivatives which requires 4 ghosts.

Reply:
I haven't really thought about how to nicely implement the 6th order derivatives. The way I have done it in my code is quite ugly (e.g. see this [PhysicsLevelClass.cpp](https://github.com/mirenradia/GRChombo-private/blob/simulation/tp_binary_bh/Examples/TPBinaryBH/TPBinaryBHLevel.cpp)). 

Maybe it's my liberal views, but if a user wants to use a large number of ghosts for whatever reason, I don't think we should stop them! However, I suspect the maximum it can be is the `block_factor` so perhaps I should check it's <= to that.

Reply:
I'm ok with that. I think most people just use the default setting now, so it's unlikely someone will request it to be 100 :-)

Reply:
Yes, I agree that most users would probably not change this away from the default.

Reply:
Done